### ISSUE TITLE
Make file operations cancellable and/or non-blocking

### ISSUE THREAD

**Issue by strugee** [nextcloud/android#366]
Problem: some file operations cannot be cancelled. This is unfortunate if you tap on the wrong one, enter an incorrect file name (rename), find the wrong folder (copy, move), etc.

Items that are currently blocking and (maybe) shouldn't be:

- [ ] Deletion
- [ ] Copying
- [ ] Renames
- [ ] Moves
- [ ] Folder creations

Not sure if these should be non-blocking. That'd be the ideal, but only if we can find a way to make the operations cancellable, IMHO.

Items that should stay blocking, but should be cancellable by hitting "back" (and _maybe_ by tapping outside of the "wait a moment" dialog? Not sure):

- [ ] Share

@jancborchardt what do you think of the above? It's pretty clear to me that the current design isn't quite optimal, but I'm not sure exactly what the best replacement would be.

**Comment by jancborchardt** [comment]
Yes, file operations should be non-blocking ideally. Especially for "simple" things like rename, the "Wait a moment" popup seems really weird, making the app seem slow. If instead we would just not show the dialog, it would likely be totally fine?

@tobiasKaminsky I don’t know if it is feasible for everything, but rename and folder creations seem light enough to make them non-blocking?

**Comment by tobiasKaminsky** [comment]
I get the intention, but technically this is harder.
Let us take folder creation.
- user clicks on "+"
- popup opens, new name specified
- folder is created
- after successful creation we go directly into this folder

How can we do this non-blocking? Assuming that it will work and pretend user is already in the new folder? But what if user then wants to create a file, but meanwhile is getting kicked out, as folder creation did not worked.

Removing files however is something we can do:
- select 5 files
- click on remove
- UI directly removes them
- on successful remove -> nothing to do
- on failure: show error and unhide failed item

Please do not put it on board, before we not have specified it.
So let us discuss what/how we can do, and if we found some specific task, we can put it on board again.

**Comment by joshtrichards** [comment]
FYI - Likely some overlap with #237 :)
